packages/core/src/stream/aisdk/v5/compat/prepare-tools.ts

```
@@ -63,11 +63,20 @@ export function prepareToolsAndToolChoice<TOOLS extends Record<string, Tool>>({
                 providerOptions: sdkTool.providerOptions,
               };
             case 'provider-defined':
+              // For provider-defined tools, extract the name from the ID to match doStream behavior
+              // ID format is typically "provider.toolName" (e.g. "openai.web_search")
+              // doStream returns just "toolName" part, so we use that as name for consistency
+              const providerId = (sdkTool as any).id;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)